### PR TITLE
Item hover tooltip

### PR DESCRIPTION
**Pull request by DavidPgl**

### CODE REVIEW

**Comment by DavidPgl** [comment]
So, @cuongtm95 and me accidentally worked on the same stuff. We can use his version of the UI-Hover as well.
Haven't done Crafting but, using this as a base, it should rather be a question of design than of logic or implementation

**Comment by cuongtm95** [comment]
Just checked out your work. Great job! Mine has the same logic but yours is for sure more detailed and has a perfect format (not to mention some other bonus stuff you added there for the inventory). Gonna delete mine and use yours! 👍 💯 

For crafting window, I find it better looking with just the icon of the item/recipe and display the tooltip on hover that shows the needed resources and some description. Or even better if we can create a new window that appears when u click on a recipe. The new window could show the item name, description, resources counters and a button "Craft" at the end. No doubt that will need more works but yeah just some thoughts.

File: Assets/Scripts/Utils/ElementInteraction/Draggable.cs [hunk 1]
```diff
@@ -11,41 +12,64 @@ namespace Utils.ElementInteraction
...
+        [Tooltip("Used for showing this on top of other UI-Elements")] [SerializeField]
+        private Canvas canvas;
+
+        [Tooltip("Used for finding UI elements with rays")] [SerializeField]
+        private GraphicRaycaster graphicRaycaster;
+
         [Tooltip("The highest parent element that will be moved")] [SerializeField]
         private Transform parent;
 
+        public GraphicRaycaster GraphicRaycaster { get; private set; }
```
Comment (Kaes3kuch3n) [inline Assets/Scripts/Utils/ElementInteraction/Draggable.cs]: You are currently using two different graphic raycasters: One which is stored in this property and fetched using GetComponentsInParent on Awake, and one which is stored in a serialised field. I guess that's not intended

**Review by Kaes3kuch3n — CHANGES_REQUESTED** [review]
Just one comment :)

File: Assets/Scripts/Utils/ElementInteraction/Draggable.cs [hunk 2]
```diff
@@ -11,41 +12,64 @@ namespace Utils.ElementInteraction
...
+            }
+        }
+
         /// <summary>
         /// Gets the GraphicRayCaster from the canvas
         /// </summary>
         private void Awake()
         {
-            // Get the closest GraphicRaycaster (in hierarchy)
-            GraphicRaycaster = GetComponentInParent<GraphicRaycaster>();
```
Comment (SamuelllG) [inline Assets/Scripts/Utils/ElementInteraction/Draggable.cs:40]: 😵 that's FORBIDDEN 😁

File: Assets/Scripts/Utils/ElementInteraction/Draggable.cs (hunk 1 above)
Comment (SamuelllG) [reply Assets/Scripts/Utils/ElementInteraction/Draggable.cs]: Hmm why do you use GetComponentsInParent anyway if you have a serialized field?

**Review by SamuelllG — CHANGES_REQUESTED** [review]
Looks good, except for the FORBIDDEN GetComponentsInParent 😝

File: Assets/Scripts/Utils/ElementInteraction/Draggable.cs (hunk 2 above)
Comment (DavidPgl) [reply Assets/Scripts/Utils/ElementInteraction/Draggable.cs:40]: I DONT CARE 🚽

File: Assets/Scripts/Utils/ElementInteraction/Draggable.cs (hunk 1 above)
Comment (DavidPgl) [reply Assets/Scripts/Utils/ElementInteraction/Draggable.cs]: Renamed it

**Review by Kaes3kuch3n — DISMISSED** [review]

**Review by Kaes3kuch3n — APPROVED** [review]

**Review by SamuelllG — APPROVED** [review]

**Pull request merged**